### PR TITLE
Reformat codebase with Ruff autoformatter and remove .ipynb checkpoints and *.pyc files

### LINKED ISSUE #5: Code style
Apply `ruff` to codebase and add automated test to ensure all subsequent code conforms to ruff standard.

### PR DESCRIPTION
**Pull request by ma595**
This PR renames the `densite.py` file to `density.py`, corrects a few spelling mistakes and applies `Ruff` autoformatter project-wide. 

- [x] Add pre-commit locally
- [x] Add remote pre-commit check that ensures formatting (eol, ruff etc) pass. 
- [x] Apply ruff project wide
- [x] Add remote ruff check
- [x] Remove .pyc and ipynb checkpoint files.    

Closes #5

### CODE REVIEW

File: requirements.txt
Comment (surbhigoel77) [inline requirements.txt:1]: Do we need to mention installation of the `pre-commit` package somewhere?

**Review by surbhigoel77 — DISMISSED** [review]
I have added a comment on using `pre-commit`. Otherwise LGTM.

**Comment by ma595** [comment]
@surbhigoel77, can you please confirm that you have installed and executed at least the `main_forcast.py` file.

File: requirements.txt
Comment (ma595) [reply requirements.txt:1]: We need a `CONTRIBUTING.md`. Let's leave this for a later commit.

**Comment by ma595** [comment]
@isaacaka can you please review this so we can get this merged.

**Review by isaacaka — DISMISSED** [review]
Looks good to merge

**Comment by ma595** [comment]
I suspect there might be annoying issues with subsequent Jupyter files when attempting to rebase any subsequent branches off of these changes. @isaacaka do you think these would be too awful to fix?

File: Notebooks/Jumper.ipynb [hunk 1]
```diff
@@ -335,24 +339,40 @@
...
+    "names = [\"zos\", \"so\", \"thetao\"]\n",
     "colors = [\"tab:blue\", \"darkgreen\", \"darkred\"]\n",
     "\n",
-    "fig = plt.figure(figsize=(6,8))\n",
+    "fig = plt.figure(figsize=(6, 8))\n",
     "for i in range(3):\n",
     "    if i == 0:\n",
-    "        plt.errorbar(np.mean(values[i]), array.deptht[0], xerr=np.std(values[i]), fmt='.', label=names[i], color=colors[i], ecolor=\"grey\")\n",
+    "        plt.errorbar(\n",
+    "            np.mean(values[i]),\n",
```
Comment (surbhigoel77) [inline Notebooks/Jumper.ipynb:350]: Is the space above this line ok?

File: Notebooks/Jumper.ipynb [hunk 2]
```diff
@@ -335,24 +339,40 @@
...
+    "            xerr=np.std(values[i]),\n",
+    "            fmt=\".\",\n",
+    "            label=names[i],\n",
+    "            color=colors[i],\n",
+    "            ecolor=\"grey\",\n",
+    "        )\n",
     "    else:\n",
-    "        plt.errorbar(np.mean(values[i], axis=1), array.deptht, xerr=np.std(values[i], axis=1), fmt='.', label=names[i], color=colors[i], ecolor=\"grey\")\n",
+    "        plt.errorbar(\n",
+    "            np.mean(values[i], axis=1),\n",
```
Comment (surbhigoel77) [inline Notebooks/Jumper.ipynb:360]: Same concern as above

File: README.md [hunk 3]
```diff
@@ -3,70 +3,70 @@ change update_rho -> thetao_new => Restart["tn"]
 # EMULATOR FRO NEMO MODEL
 
 ![img1](img/emulator.png)
-Add retstart and prepare 
+Add retstart and prepare
```
Comment (surbhigoel77) [inline README.md:6]: retstart?

File: Notebooks/Jumper.ipynb (hunk 1 above)
Comment (ma595) [reply Notebooks/Jumper.ipynb:350]: What space?

**Review by surbhigoel77 — APPROVED** [review]
Have left a couple of minor comments. Otherwise LGTM.

File: Notebooks/Jumper.ipynb (hunk 1 above)
Comment (surbhigoel77) [reply Notebooks/Jumper.ipynb:350]: My bad, I just saw the gap in the github window, there is not actual space (saw the line number now and yes there is no actual space)

**Review by surbhigoel77 — APPROVED** [review]
LGTM

**Pull request merged**